compiler/src/iree/compiler/Codegen/LLVMCPU/test/pipeline_tests.mlir

```diff
@@ -602,3 +602,103 @@ hal.executable private @ukernel_pass_through {
 //  CHECK-SAME:         ins(%[[SUBVIEW_INPUT0]], %[[SUBVIEW_INPUT1]]
 //  CHECK-SAME:         outs(%[[SUBVIEW_OUTPUT]]
 
+// -----
+
+// Check Armv9 Streaming SVE mode is enabled for the following pipelines:
+//
+//   * CPUBufferOpsTileAndVectorize
+//   * CPUDoubleTilingPeelingExpert
+//   * CPUConvTileAndDecomposeExpert
+
+#pipeline_layout = #hal.pipeline.layout<push_constants = 2, sets = [
+  #hal.descriptor_set.layout<0, bindings = [
+    #hal.descriptor_set.binding<0, storage_buffer>,
+    #hal.descriptor_set.binding<1, storage_buffer>,
+    #hal.descriptor_set.binding<2, storage_buffer>
+  ]>
+]>
+
+#executable_target_embedded_elf_arm_64_ = #hal.executable.target<"llvm-cpu", "embedded-elf-arm_64", {
+  cpu_features = "+sve,+sme",
+  data_layout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128",
+  native_vector_size = 16 : index,
+  target_triple = "aarch64-unknown-unknown-eabi-elf"
+}>
+
+hal.executable private @aarch64_ssve__cpu_buffer_ops_tile_and_vectorize {
+  hal.executable.variant public @embedded_elf_arm_64, target = #executable_target_embedded_elf_arm_64_ {
+    hal.executable.export public @dispatch ordinal(0) layout(#pipeline_layout) attributes {
+      translation_info = #iree_codegen.translation_info<CPUBufferOpsTileAndVectorize>
+    } {
+    ^bb0(%arg0: !hal.device, %arg1: index, %arg2: index):
+      hal.return %arg1, %arg2, %arg2 : index, index, index
+    }
+    builtin.module {
+      func.func @dispatch() { return }
```

Comment:
Currently hitting a problem with these tests as they go through the different experts but there is no linalg op and, therefore, we don't generate any lowering config. A couple of suggestions:
  1. Could we add at least one linalg operation?
  2. Would it make sense to move the code that adds the SME pass to a common place for all the pipelines? Perhaps to the LowerToLLVM pass or at the end of LLVMCPULowerExecutableTargetPass::runOnOperation?
  ```
    if (enableAArch64SSVE)
    nestedModulePM.addNestedPass<func::FuncOp>(
        mlir::arm_sme::createEnableArmStreamingPass(
            mlir::arm_sme::ArmStreaming::Locally));
  ```
  to a common place for all the pipelines? Perhaps to the LowerToLLVM pass or at the end of LLVMCPULowerExecutableTargetPass::runOnOperation?

Reply:
Hi Diego, I'm at a conference today but I'll try take a look later this afternoon.

Reply:
The problem is that the dispatches in these tests are empty and it looks like our pipelines haven't been taught to deal with that case. I guess it makes sense because if the dispatch is empty it shouldn't have been created. We can just add a dummy linalg op to the tests to fix the problem.

Regarding the location of the SSVE pass, you know better what else will be needed but, currently, it is just about introducing the attributes at function level after all the pipeline transformations has happen. That looks like generic to me and not so much pipeline specific so perhaps moving it to `addLowerToLLVMPasses` makes sense. We have plenty of post-pipeline transformations there. Since the SSVE pass is under a flag, it should be fine but, again, you know better :). This is just a suggestion. Nothing to worry about right now if we add a dummy linalg op to the tests.